src/ground_truth_nsrts.py

```diff
@@ -749,6 +749,42 @@ def openlid_sampler(state: State, rng: np.random.Generator,
                         set(), option, option_vars, openlid_sampler)
     nsrts.add(openlid_nsrt)
 
+    # PlaceOnTable (from somewhere else on the table)
+    for holding_rot_pred in [HoldingSide, HoldingTop]:
+        obj = Variable("?obj", obj_type)
+        robot = Variable("?robot", robot_type)
+        parameters = [obj, robot]
+        option_vars = [robot]
+        option = Place
+        preconditions = {
+            LiftedAtom(Holding, [obj]),
+            LiftedAtom(holding_rot_pred, [robot]),
+            LiftedAtom(OnTable, [obj]),
+        }
+        add_effects = {
+            LiftedAtom(GripperOpen, [robot]),
+        }
+        delete_effects = {
+            LiftedAtom(Holding, [obj]),
+            LiftedAtom(holding_rot_pred, [robot]),
+        }
+
+        def placeontable_sampler(state: State, rng: np.random.Generator,
+                                 objs: Sequence[Object]) -> Array:
+            # Always release the object where it is, to avoid the possibility
```

Comment:
I see. So you're avoiding the EnvironmentFailure above. I guess we're hoping that a learned sampler would have this same behavior?

Reply:
yeah